pennylane/ops/qubit/parametric_ops_multi_qubit.py

```diff
@@ -595,6 +595,149 @@ def _pauli_rot_resources(pauli_word):
 add_decomps(PauliRot, pauli_rot_decomposition)
 
 
+def decomp_int_to_powers_of_two(k: int, n: int) -> list[int]:
```

Comment:
When reviewing, it might be easiest to first look at the example I commented on the tests for this function :)